_posts/2019-08-15-release-1.9.0.md

```diff
@@ -0,0 +1,352 @@
+
+Up to Flink 1.9, accessing the state of a job from the outside was limited to
+the (still) experimental [Queryable
+State](https://ci.apache.org/projects/flink/flink-docs-stable/dev/stream/state/queryable_state.html).
+This release introduces a new, powerful library to read, write and modify
+state snapshots using the batch DataSet API. In practice, this means:
+
+ * Flink job state can be bootstrapped by reading data from external systems,
+   such as external databases, and converting it into a savepoint.
+ * State in savepoints can be queried using any of Flink’s batch APIs
+   (DataSet, Table, SQL), for example to analyze relevant state patterns or
+   check for discrepancies in state that can support application auditing or
+   troubleshooting.
+ * The schema of state in savepoints can be migrated offline, compared to the
+   previous approach requiring online migration on schema access.
+ * Invalid data in savepoints can be identified and corrected.
+
+The new State Processor API covers all variations of snapshots: savepoints,
+full checkpoints and incremental checkpoints. More details are available in
+[FLIP-43](https://cwiki.apache.org/confluence/display/FLINK/FLIP-43%3A+State+Processor+API)
+
+
+### Stop-with-Savepoint (FLIP-34)
+
+[Cancelling with a
+savepoint](https://ci.apache.org/projects/flink/flink-docs-stable/ops/state/savepoints.html#operations)
+is a common operation for stopping/restarting, forking or updating Flink jobs.
+However, the existing implementation did not guarantee output persistence to
+external storage systems for exactly-once sinks. To improve the end-to-end
+semantics when stopping a job, Flink 1.9 introduces a new `SUSPEND` mode to
+stop a job with a savepoint that is consistent with the emitted data.
+You can suspend a job with Flink’s CLI client as follows:
+
+```
+bin/flink stop -s [:targetDirectory] :jobId
+```
+
+The final job state is set to `FINISHED` on success, allowing
+users to detect failures of the requested operation. 
+
+More details are available in
+[FLIP-34](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=103090212)
+
+
+### Flink WebUI Rework
+
+After a
+[discussion](http://apache-flink-mailing-list-archive.1008284.n3.nabble.com/DISCUSS-Change-underlying-Frontend-Architecture-for-Flink-Web-Dashboard-td24902.html)
+about modernizing the internals of Flink’s WebUI, this component was
+reconstructed using the latest stable version of Angular — basically, a bump
+from Angular 1.x to 7.x. The redesigned version is the default in 1.9.0,
+however there is a link to switch to the old WebUI.
+
+<div class="row"> <div class="col-sm-6"> <span><img class="thumbnail"
+    src="{{site.baseurl}}/img/blog/release-19-web1.png" /></span> </div> <div
+    class="col-sm-6"> <span><img class="thumbnail"
+    src="{{site.baseurl}}/img/blog/release-19-web2.png" /></span> </div>
+    </div>
+
+**Note:** Moving forward, we will not guarantee feature parity for the old
```

Comment:
Replace "we ..." with "feature parity for the old version of the WebUI will not be guaranteed".